config/config.go

```diff
@@ -61,6 +61,25 @@ type AnalysisConfig struct {
 
 // Validate validates the analysis configuration.
 func (cfg *AnalysisConfig) Validate() error {
+	sourceConfigured := false
+	if cfg.Source.History != nil {
+		sourceConfigured = true
+		if cfg.Source.History.ChainName == "" {
+			return fmt.Errorf("source.history.chian_name not provided")
```

Comment:
nit: typo

Reply:
oo good find

Reply:
rewrote the validation function when I changed the structure